---
title: "Abundance and Distribution of Pseudo-nitzschia in the Global Ocean: V4 exploration"
author: "Lucia Campese"
date: "July 2020"
output:
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

<style>
.column-left{
  float: left;
  width: 50%;
  text-align: left;
}
.column-right{
  float: right;
  width: 50%;
  text-align: right;
}
</style>


<br>

### Dataset
##### Omic data: TARA Oceans and Tara Oceans Polar circle metabarcording (V4)
##### Taxon: Pseudonitzschia
##### Geographic scale: Global


```{r eval=FALSE}

#FALKOR

#As adviced by Nico: work with sequences with at least 3 reads
distribution <- fread("18S_V4_1191_samples.distr")

dim(distribution)
[1] 221040090         3

colnames(distribution) <- c("seqId", "sample", "counts")

#1. select only counts > 2
distribution_filtered <- distribution %>% 
  filter(counts>2)

dim(distribution_filtered)
#[1] 12229521        3


taxonomy <- fread("18S_V4_1191_samples.results")

dim(taxonomy)
#[1] 10547358        5

colnames(taxonomy) <- c("seqId", "tot.numb.reads", "perc.identity", "lineage", "idpr2")


##check for pseudonitzschia
Pn_tax <- taxonomy[grep("Pseudo-nitzschia", taxonomy$lineage),]

dim(Pn_tax)
#[1] 30946     5

save(Pn_tax, file = "Pn_tax.RData")


Pn_distr <- distribution %>% 
  filter(seqId %in% Pn_tax$seqId)

save(Pn_distr, file = "Pn_distr.RData")


Pn_seqId<- Pn_tax %>% 
  select(OTU)

write.table(Pn_seqId, file = "Pn_sequenceId.txt", row.names=F, col.names=F, quote = F)



#lcampese@falkor-frontend:~/Data/metaB/V4$ grep -w -A 1 -Ff Pn_sequenceId.txt 18S_V4_1191_samples.fas --no-group-separator > Pn_V4_seq.fas
#this command subsets the total fastas with the Id of pseudonitzschia sequences.
#NOTA: l'ultima sequenza rimane troncata. (qua sto usando -A 1 quindi prende solo una riga dopo l'header corrispondente, per? usando -A 2 lui 
#mi d? quasi sempre header + sequenza + riga sotto (quindi un altro header che non serve a un cazzo)). Il problema non ? risolto ma probabilmente non ? 
#rilevante (in ogni caso noi faremo blast sul totale).


#creo subset per Roberta e le mando subset in .fas e lista dell'annotazione tassonomica in .csv
#grep "^>" subset_PnV4.fas| sed -e 's/>//' -e 's/;/,/g' > subset_PnV4_headers.txt
```





```{r warning=FALSE, message=FALSE, cache=TRUE}
load("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/Pn_tax.RData")

library(tidyverse)
library(gridExtra)
library(kableExtra)

Percid <- 
  Pn_tax %>% 
  select(perc.identity) %>% 
  summary()


Pn_tax %>% 
  ggplot() +
  aes(x = perc.identity) +
  geom_histogram(bins = 100) +
  theme_bw() +
  ggtitle("Percentage of identity to the PR2 reference", paste0("Pseudonitzschia, TARA V4; ", n_distinct(Pn_tax$seqId), " distinct amplicons")) +
  annotation_custom(tableGrob(Percid), xmin=65, xmax=90, ymin=3000, ymax=9000)

table_percid_amplicons <- Pn_tax %>% 
  mutate(perc.identity_rounded = round(perc.identity),
         `percentage of identity` = case_when(perc.identity_rounded < 97 ~ "< 97",
                          perc.identity_rounded  == 97 ~ "97",
                          perc.identity_rounded  == 98 ~ "98",
                          perc.identity_rounded  == 99 ~ "99",
                          perc.identity_rounded  == 100 ~ "100")) %>% 
  group_by(`percentage of identity`) %>% 
  summarise(`distinct amplicons` = n_distinct(seqId)) %>% 
  arrange(desc(as.numeric(`percentage of identity`)))

table_percid_amplicons %>%
  kable() %>%
  kable_styling()
```

Select only Pseudonitzschia with 100% of identity with the reference and calculate abundance and richness per each station, where

- Abundance = total number of read counts of all amplicons;
- Richness = number of different amplicons

*so far no filtering on size fraction nor depth.

```{r warning=FALSE, message=FALSE, cache=TRUE, fig.width=12, fig.height==10}

load("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/Pn_distr_stations_100.RData")
library(maps)

world_map <- map_data("world")
p <- ggplot() + coord_fixed()

base_world3 <- p + geom_polygon(data=world_map,aes(x=long, y=lat,group=group),colour="gray80",size=0.2, fill="gray80")+
  theme_bw() + coord_fixed()


Pn_abric <- Pn_distr_stations_100 %>% 
  group_by(Station.label) %>% 
  summarise(abundance = sum(counts), richness = n_distinct(seqId),
            Latitude = mean(Latitude), Longitude = mean(Longitude)) %>% 
  ungroup()

#map
base_world3+
  geom_point(data=Pn_abric,
             aes(x = Longitude, y = Latitude, 
                 size = abundance,
                 fill = richness), pch = 21, col = "black") +
  scale_fill_viridis_c() +
  theme_minimal() +
  theme(legend.text=element_text(size=10)) +
  ggtitle("Pseudo-nitzschia abundance and richness per station (V4)",
          "Identity 100%") 


Pn_abric %>%
  mutate(Station = substr(Station.label, 6, 8)) %>% 
  select(Station, abundance, richness) %>% 
  pivot_longer(abundance:richness) %>% 
  ggplot() +
  aes(x=Station, y = value) +
  geom_col() +
  theme_bw() +
  theme(axis.text.x = element_text(angle=50, size=6, vjust = 0.5)) +
  facet_wrap(~ name, scales = "free", nrow = 2) 

```


```{r warning=FALSE, message=FALSE, cache=TRUE}
library(kableExtra)

abric <- as.data.frame(rbind(summary(Pn_abric$abundance), summary(Pn_abric$richness)))

rownames(abric) <- c("abundance", "richness")

abric %>% kable() %>% 
  kable_styling()


Pn_abric %>%
  mutate(Station = substr(Station.label, 6, 8)) %>% 
  select(Station, abundance, richness) %>% 
  pivot_longer(abundance:richness) %>% 
  ggplot() +
  aes(x=value) +
  geom_histogram() +
  theme_bw() +
  facet_wrap(~ name, scales = "free") 


Pn_abric %>% 
  mutate(Station = substr(Station.label, 6, 8)) %>% 
  ggplot() +
  aes(x = richness, y = abundance) +
  theme_bw() +
  geom_point() +
  #scale_x_log10("log-scaled richness") +
  #scale_y_log10("log-scaled abundance") +
  ggtitle("Relation between Pseudo-nitzschia abundance and richness per station (V4)")
```

