---
title: "*Pseudonitzschia* and its eukaryotic symbionts"
subtitle: "Global scale co-occurrence analysis"
author: "Lucia Campese"
date: "September 2020"
output:
  html_document:
    code_folding: hide
---
---
  
  ```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

<style>
.column-left{
  float: left;
  width: 50%;
  text-align: left;
}
.column-right{
  float: right;
  width: 50%;
  text-align: right;
}
</style>
  
<br>


### Dataset
##### Data: TARA Oceans and Tara Oceans Polar circle metabarcording (V4)
##### Taxa: *Pseudonitzschia*, Fungi, Oomycetes
##### Geographic scale: Global


<br>

#### **Building datasets**

<br>

##### 1. *Pseudonitzschia*

The used dataset is made of **324** different **sequences** assigned to *Pseudonitzschia* species with a **100% identity** to MareChiara database and occurring in at least **3 sampling stations** in the TARA Oceans dataset. 

```{r warning=FALSE, message=FALSE, cache=TRUE}

library(data.table)
library(tidyverse)

Pn_candi_identity100.distr <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/cfr100pid/18S_V4_1191_samples_Pncandi_3readsmin_pid100.distr") 

colnames(Pn_candi_identity100.distr) <- c("seqId", "sample", "readcounts")

paste0(n_distinct(Pn_candi_identity100.distr$seqId), " Pseudonitzschia sequence Ids")

Pn_candi_identity100.tax <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/cfr100pid/Pn_tara_candi_final_pid100.txt") %>%
  separate(V1, into = c("seqId", "size"), sep = ";") %>% 
  rename(tax = V2, pid = V3) %>% 
  select(seqId, tax, pid)

#subset tax (still with 508 seq) using the distr file (with the corret 324 seq) {the aim is to add the taxonomy info}

Pn_candi_identity100.distr.tax <- inner_join(Pn_candi_identity100.tax, Pn_candi_identity100.distr, by = "seqId")

##add env

env <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/18S_V4_1191_samples_context.tsv")

Pn_candi_identity100.distr.tax.env <- inner_join(Pn_candi_identity100.distr.tax, env) %>% 
  select(seqId:readcounts,Pangaea_id:minimum.generation.time.h)


final_tara_mc <- Pn_candi_identity100.distr.tax.env %>% 
  mutate(species = case_when(grepl("allochrona", tax) ~ "allochrona",
                             grepl("americana", tax) ~ "ID2_amer|cacia|circump|granii",
                             grepl("arenysensis", tax) ~ "ID1_aren|dolo|arc|subcur",
                             grepl("ID3_P._australis", tax) ~ "ID3_australis|multiseries",
                             grepl("P._australis_SPC21", tax) ~ "australis_single",
                             grepl("batesiana", tax) ~ "batesiana",
                             grepl("brasiliana", tax) ~ "ID4_brasiliana|lineola",
                             grepl("calliantha", tax) ~ "calliantha",
                             grepl("cuspidata", tax) ~ "cuspidata",
                             grepl("decipiens", tax) ~ "decipiens",
                             grepl("delicatissima", tax) ~ "delicatissima",
                             grepl("fraudulenta", tax) ~ "fraudulenta",
                             grepl("fukuyoi", tax) ~ "fukuyoi",
                             grepl("galaxiae", tax) ~ "galaxiae",
                             grepl("heimii", tax) ~ "heimii",
                             grepl("kodamae", tax) ~ "ID5_kodamae|lundholmiae",
                             grepl("mannii", tax) ~ "mannii",
                             grepl("micropora", tax) ~ "micropora",
                             grepl("P._multiseries_NparI", tax) ~ "multiseries_single",
                             grepl("multistriata", tax) ~ "multistriata",
                             grepl("pseudodelicatissima", tax) ~ "pseudodelicatissima",
                             grepl("pungens", tax) ~ "pungens",
                             grepl("sabit", tax) ~ "sabit",
                             grepl("turgidula", tax) ~ "turgidula",
                            TRUE ~ "ERRORE")) %>% 
  mutate(length = "tobedone") %>% 
  select(seqId, length, species, sample, readcounts, Station.label:Depth.chloro.max) %>%
  mutate(dataset = c("TARA-MC"))
```

<br>

<br>


##### 2. Fungi

The used dataset is made of **34523** different **sequences** assigned to *Fungi* with a variable percentage of identity to PR2 database and occurring in at least **3 sampling stations** in the TARA Ocean dataset. 
 

```{r warning=FALSE, message=FALSE, cache=TRUE}

library(data.table)
library(tidyverse)

Fungi.distr <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/Symbionts/18S_V4_1191_samples_Fungi_3readsmin.distr")

Fungi.tax <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/Symbionts/18S_V4_1191_samples_Fungi_3readsmin.results", header = F, sep = "\t") #17 sequences


colnames(Fungi.distr) <- c("seqId", "sample", "readcounts")
colnames(Fungi.tax) <- c("seqId", "totreads", "perc.identity", "lineage", "idpr2")


#Fungi.tax %>% 
#  splitstackshape::cSplit("lineage", "|") %>%
#  filter(lineage_06 != "Fungi")

##checked sono tutti funghi
```
<div class="column-left">
```{r warning=FALSE, message=FALSE, cache=TRUE}

Fungi_percid <- Fungi.tax %>% 
  select(perc.identity) %>% 
  summary()


Fungi.tax %>% 
  ggplot + 
  aes(x = perc.identity, 
      fill = totreads) + 
  geom_histogram(bins = 100) + 
  theme_bw() + 
  # xlim(90,100) +
  annotation_custom(gridExtra::tableGrob(Fungi_percid), xmin=80, xmax=90, ymin=6000, ymax=12000) +
  ggtitle("Percentage of identity to the PR2 reference", 
          subtitle= paste0("Fungi, TARA V4; ", 
                           n_distinct(Fungi.tax$seqId), 
                           " distinct amplicons"))
```
</div><div class="column-right">
```{r warning=FALSE, message=FALSE, cache=TRUE}

Fungi.distr.tax <- inner_join(Fungi.distr, Fungi.tax, by = "seqId") %>%
  select(-totreads)


env <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/18S_V4_1191_samples_context.tsv")

Fungi.distr.tax.env <- inner_join(Fungi.distr.tax, env)


Fungi.occurrences <- Fungi.distr.tax.env %>% 
  group_by(seqId) %>% 
  summarise(Occurrence = n_distinct(Station.label)) %>% 
  arrange(Occurrence) 


table_occurrences <- Fungi.occurrences %>% 
  select(Occurrence) %>% 
  summary()


Fungi.occurrences %>% 
  ggplot() +
  aes(x=Occurrence) +
  geom_histogram(bins = 100) +
  theme_bw() +
  annotation_custom(gridExtra::tableGrob(table_occurrences), xmin=25, xmax=55, ymin=7500, ymax=10000) +
  ggtitle("Occurrence of sequences along stations", 
          subtitle= paste0("Fungi, TARA V4; ", 
                           n_distinct(Fungi.occurrences$seqId), 
                           " distinct amplicons"))
```
</div>

```{r warning=FALSE, message=FALSE, cache=TRUE}

##filter on occurrence

Fungi.distr.tax.env.filtered <- Fungi.distr.tax.env %>% 
  group_by(seqId) %>% 
  mutate(Occurrence = n_distinct(Station.label)) %>% 
  ungroup() %>% 
  filter(Occurrence > 1)

Fungi.distr.tax.env1 <- Fungi.distr.tax.env.filtered %>% 
  select(seqId, lineage, perc.identity, sample, readcounts, Station.label:Depth.chloro.max)


#Fungi.distr.tax.env1 is the final file with seqIds, taxonomical annotation, distribution and environmental info


paste0(n_distinct(Fungi.distr.tax.env1$seqId), " Fungi sequences")


Fungi_3reads2stationsmin_seqid <- Fungi.distr.tax.env1 %>% 
  select(seqId) %>% 
  unique()

write.csv(Fungi_3reads2stationsmin_seqid, "Fungi_3reads2stationsmin_seqid.csv", row.names = F)


###levo il nome della colonna su excel e esporto il file come txt file e lo porto su falkor 
```

```{r, eval=FALSE}

###FALKOR
###take fastas

#wc -l 34523 Fungi_3reads2stationsmin_seqid.txt

#tuttapposto mi trovo 

##solo che il mio file ? un ascii with crlf terminators e il fasta no, quindi prima devo convertire il file in formato decente

cat Fungi_3reads2stationsmin_seqid.txt | tr -d '\r' > Fungi_3reads2stationsmin_seqid_ascii.txt

#use seqIds  to retrieve fasta of the sequences
srun grep -w -A 1 -Ff Symbionts/Fungi_3reads2stationsmin_seqid_ascii.txt 18S_V4_1191_samples.fas --no-group-separator > 18S_V4_1191_samples_Fungi_3reads2stationsmin.fas &
  
#check if it retrieved the right number of sequences

grep ">" 18S_V4_1191_samples_Fungi_3reads2stationsmin.fas -c
#34523
wc -l Fungi_3reads2stationsmin_seqid_ascii.txt
#34523

#tutt'appost. 


##quindi ora ho sia i fasta: 18S_V4_1191_samples_Fungi_3reads2stationsmin.fas
##che la tabella con le info sulla tassonomia parametri ambientali ecc: Fungi.distr.tax.env1
```

<br>

##### 3. Oomycetes

The used dataset is made of **2997** different **sequences** assigned to Peronosporomycetes with a variable percentage of identity to PR2 database and occurring in at least **3 sampling stations** in the TARA Ocean dataset. 

```{bash, eval = FALSE}
#1. Extract Peronosporomycetes  sequences from the the V4

srun cat 18S_V4_1191_samples.results | grep Peronosporomycetes  > 18S_V4_1191_samples_oomyc.results &
  
  
#take only sequences with at least 3 reads (seen three times in the whole dataset) [*]
srun awk '$2>2' 18S_V4_1191_samples_oomyc.results > 18S_V4_1191_samples_oomyc_3readsmin.results &
  
#wc -l 5403 18S_V4_1191_samples_oomyc.results
#wc -l 4451 18S_V4_1191_samples_oomyc_3readsmin.results
  
#[*]as adviced by Nicholas Henry.
  

##SHOULD I TAKE 100% HERE? NON PENSO   
  
#Extract only seq ids (column 1)
awk '{print $1}' 18S_V4_1191_samples_oomyc_3readsmin.results > 18S_V4_1191_samples_oomyc_3readsmin_seqid.txt

#wc -l: 4451 18S_V4_1191_samples_oomyc_3readsmin_seqid.txt


#use the retrieved 4451 sequence Ids to subset the distr file (i.e. the one 
#containing seqId|sampleId|readcounts

srun grep -Ff 18S_V4_1191_samples_oomyc_3readsmin_seqid.txt ~/Data/metaB/V4/18S_V4_1191_samples.distr > 18S_V4_1191_samples_oomyc_3readsmin.distr &
  

```

```{r warning=FALSE, message=FALSE, cache=TRUE}
library(data.table)
library(tidyverse)

oomyc.distr <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/Symbionts/18S_V4_1191_samples_oomyc_3readsmin.distr")


oomyc.tax <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/Symbionts/18S_V4_1191_samples_oomyc_3readsmin.results", header = F, sep = "\t") #17 sequences


colnames(oomyc.distr) <- c("seqId", "sample", "readcounts")
colnames(oomyc.tax) <- c("seqId", "totreads", "perc.identity", "lineage", "idpr2")

#oomyc.tax %>% 
  #splitstackshape::cSplit("lineage", "|")




##checked sono tutti funghi
```
<div class="column-left">
```{r warning=FALSE, message=FALSE, cache=TRUE}

oomyc_percid <- oomyc.tax %>% 
  select(perc.identity) %>% 
  summary()


oomyc.tax %>% 
  ggplot + 
  aes(x = perc.identity, 
      fill = totreads) + 
  geom_histogram(bins = 100) + 
  theme_bw() + 
  # xlim(90,100) +
  annotation_custom(gridExtra::tableGrob(oomyc_percid), xmin=80, xmax=90, ymin=6000, ymax=12000) +
  ggtitle("Percentage of identity to the PR2 reference", 
          subtitle= paste0("Oomycetes, TARA V4; ", 
                           n_distinct(oomyc.tax$seqId), 
                           " distinct amplicons"))
```
</div><div class="column-right">
```{r warning=FALSE, message=FALSE, cache=TRUE}

oomyc.distr.tax <- inner_join(oomyc.distr, oomyc.tax, by = "seqId") %>% 
  select(-totreads)

env <- fread("C:/Users/Userszn/Documents/PhD/script/Pseudonitzschia/metaB/V4/18S_V4_1191_samples_context.tsv")

oomyc.distr.tax.env <- inner_join(oomyc.distr.tax, env)


oomyc.occurrences <- oomyc.distr.tax.env %>% 
  group_by(seqId) %>% 
  summarise(Occurrence = n_distinct(Station.label)) %>% 
  arrange(Occurrence) 

oomyc.table_occurrences <- oomyc.occurrences %>% 
  select(Occurrence) %>% 
  summary()

oomyc.occurrences %>% 
  ggplot() +
  aes(x=Occurrence) +
  geom_histogram(bins = 100) +
  theme_bw() +
  annotation_custom(gridExtra::tableGrob(oomyc.table_occurrences), xmin=25, xmax=55, ymin=7500, ymax=12000) +
  ggtitle("Occurrence of sequences along stations", 
          subtitle= paste0("Oomycetes, TARA V4; ", 
                           n_distinct(oomyc.occurrences$seqId), 
                           " distinct amplicons"))
```
</div>
```{r warning=FALSE, message=FALSE, cache=TRUE, fig.align='center'}
##filter on occurrence

oomyc.distr.tax.env.filtered <- oomyc.distr.tax.env %>% 
  group_by(seqId) %>% 
  mutate(Occurrence = n_distinct(Station.label)) %>% 
  ungroup() %>% 
  filter(Occurrence > 1)

oomyc.distr.tax.env1 <- oomyc.distr.tax.env.filtered %>% 
  select(seqId, lineage, perc.identity, sample, readcounts, Station.label:Depth.chloro.max)


#Fungi.distr.tax.env1 is the final file with seqIds, taxonomical annotation, distribution and environmental info


paste0(n_distinct(oomyc.distr.tax.env1$seqId), " Oomycetes sequences")


oomyc_3reads2stationsmin_seqid <- oomyc.distr.tax.env1 %>% 
  select(seqId) %>% 
  unique()

write.csv(oomyc_3reads2stationsmin_seqid, "oomyc_3reads2stationsmin_seqid.csv", row.names = F)


###levo il nome della colonna su excel e esporto il file come txt file e lo porto su falkor 
```

```{r eval=FALSE}

###FALKOR
###take fastas

#wc -l 2997 oomyc_3reads2stationsmin_seqid.txt

#tuttapposto mi trovo 

##solo che il mio file ? un ascii with crlf terminators e il fasta no, quindi prima devo convertire il file in formato decente

cat oomyc_3reads2stationsmin_seqid.txt | tr -d '\r' > oomyc_3reads2stationsmin_seqid_ascii.txt

#use seqIds  to retrieve fasta of the sequences


srun grep -w -A 1 -Ff oomyc_3reads2stationsmin_seqid_ascii.txt ~/Data/metaB/V4/18S_V4_1191_samples.fas --no-group-separator > 18S_V4_1191_samples_oomyc_3reads2stationsmin.fas &
  
#check if it retrieved the right number of sequences

grep ">" 18S_V4_1191_samples_oomyc_3reads2stationsmin.fas -c
#2997

wc -l oomyc_3reads2stationsmin_seqid_ascii.txt
#2997

#tutt'appost. 

##quindi ora ho sia i fasta: 18S_V4_1191_samples_oomyc_3reads2stationsmin.fas
##che la tabella con le info sulla tassonomia parametri ambientali ecc: oomyc.distr.tax.env1
```


```{r warning=FALSE, message=FALSE, cache=TRUE, fig.height=10, fig.width=12}

library(maps)

world_map <- map_data("world")
p <- ggplot() + coord_fixed()

base_world3 <- p + geom_polygon(data=world_map,aes(x=long, y=lat,group=group),colour="gray80",size=0.2, fill="gray80")+
  theme_bw() + coord_fixed()


#map
oomyc.map <- oomyc.distr.tax.env1 %>% 
  group_by(Station.label) %>% 
  mutate(Lat= mean(Latitude),
         Long=mean(Longitude),
         abundance.per.station = sum(readcounts),
         richness.per.station.amplicons = n_distinct(seqId)) %>%
  ungroup() %>% 
  select(seqId, Station.label, Lat, Long, abundance.per.station,
         richness.per.station.amplicons) %>% unique()

base_world3+
  geom_point(data=oomyc.map,
             aes(x = Long, y = Lat, 
                 size = abundance.per.station,
                 fill = richness.per.station.amplicons),
             pch = 21, col = "black") +
  scale_fill_viridis_c() +
  theme_minimal() +
  theme(legend.text=element_text(size=10)) +
  ggtitle("Oomycetes abundance and richness per station (V4)", "") 


####map
fungi.map <- Fungi.distr.tax.env1 %>% 
  group_by(Station.label) %>% 
  mutate(Lat= mean(Latitude),
         Long=mean(Longitude),
         abundance.per.station = sum(readcounts),
         richness.per.station.amplicons = n_distinct(seqId)) %>%
  ungroup() %>% 
  select(seqId, Station.label, Lat, Long, abundance.per.station,
         richness.per.station.amplicons) %>% unique()

base_world3+
  geom_point(data=fungi.map,
             aes(x = Long, y = Lat, 
                 size = abundance.per.station,
                 fill = richness.per.station.amplicons),
             pch = 21, col = "black") +
  scale_fill_viridis_c() +
  theme_minimal() +
  theme(legend.text=element_text(size=10)) +
  ggtitle("Fungi abundance and richness per station (V4)", "") 






tbm_pseudonitzschia <- final_tara_mc %>% 
  select(seqId, sample:Depth.chloro.max) %>% 
  mutate(taxogroup = "Pseudonitzschia")

tbm_fungi <- Fungi.distr.tax.env1 %>% 
  select(seqId, sample:Depth.chloro.max) %>% 
  mutate(taxogroup = "Fungi")

tbm_oomyc <- oomyc.distr.tax.env1 %>% 
  select(seqId, sample:Depth.chloro.max) %>% 
  mutate(taxogroup = "Oomycetes") 



pn_oomyc_fungi.map <- rbind(tbm_pseudonitzschia, tbm_oomyc, tbm_fungi) %>% 
  group_by(Station.label) %>% 
  mutate(Lat= mean(Latitude),
         Long=mean(Longitude)) %>% 
  ungroup() %>% 
  group_by(Station.label, taxogroup) %>% 
  mutate(abundance.per.station = sum(readcounts),
         richness.per.station.amplicons = n_distinct(seqId)) %>%
  ungroup() %>% 
  select(seqId, Station.label, Lat, Long, abundance.per.station,
         richness.per.station.amplicons, taxogroup) %>% unique()

base_world3+
  geom_point(data=pn_oomyc_fungi.map,
             aes(x = Long, y = Lat, 
                 size = abundance.per.station,
                 fill = taxogroup),
             pch = 21, col = "black") +
  #scale_fill_viridis_c() +
  theme_minimal() +
  theme(legend.text=element_text(size=10)) +
  ggtitle("Abundance and richness per station (V4)", "") 
```

